crates/sui-node/src/lib.rs

```diff
@@ -446,17 +446,42 @@ impl SuiNode {
         epoch_store: Arc<AuthorityPerEpochStore>,
         checkpoint_store: Arc<CheckpointStore>,
         state_sync_handle: state_sync::Handle,
+        p2p_network: Network,
         registry_service: &RegistryService,
     ) -> Result<ValidatorComponents> {
+        let system_state = state
+            .get_sui_system_state_object()
+            .expect("Reading Sui system state object cannot fail");
+        // Spawn connectivity monitor for network
+
+        let peer_ids = system_state.get_current_epoch_peer_ids();
+
+        let peer_ids_to_authority_names =
+            system_state.get_current_epoch_peer_id_to_authority_names();
+
+        let (connection_status_sender, connection_status_receiver) = channel(DEFAULT_CHANNEL_SIZE);
+
+        let network_connection_metrics =
+            NetworkConnectionMetrics::new("sui", &registry_service.default_registry());
+        let _connection_monitor_handle = narwhal_network::connectivity::ConnectionMonitor::spawn(
+            p2p_network.downgrade(),
+            network_connection_metrics,
+            peer_ids,
+            Some(connection_status_sender),
+        );
```

Comment:
Probably not a comment related that much to this PR, but as an overall observation, we might need to consider the reconfiguration case with committee change. I am not aware what's the plan for networking around this (ex shutting down network? adding/removing peers from the `p2p_network` ? ) , because the connectivity monitor might also need to get updated. 
CC @bmwill

Reply:
Yeah we'll need to, at least for validators, update the known peers in order to ensure validators in the new epoch have whitelisted each other